Sources/Brave/Frontend/Browser/Playlist/BasicAuthCredentials.swift

```diff
@@ -5,44 +5,87 @@
 
 import Foundation
 import UIKit
+import BraveCore
+import os.log
 
 class BasicAuthCredentialsManager: NSObject, URLSessionDataDelegate {
-  private let domains: [String]
+  private static var credentials = [String: URLCredential]()  // ["origin:port": credential]
   
-  init(for domains: [String]) {
-    self.domains = domains
+  public static let validDomains: Set<String> = [
+    // Brave Search
+    "search.brave.com", "search.brave.software",
+    "search.bravesoftware.com", "safesearch.brave.com",
+    "safesearch.brave.software", "safesearch.bravesoftware.com",
+    "search-dev-local.brave.com",
+    
+    // Playlist
+    "playlist.bravesoftware.com", "playlist.brave.com"
+  ]
+  
+  static func setCredential(origin: String, credential: URLCredential?) {
+    credentials[origin] = credential
   }
 
   func urlSession(
     _ session: URLSession,
     didReceive challenge: URLAuthenticationChallenge
   ) async -> (URLSession.AuthChallengeDisposition, URLCredential?) {
-    guard challenge.protectionSpace.authenticationMethod == NSURLAuthenticationMethodHTTPBasic ||
-            challenge.protectionSpace.authenticationMethod == NSURLAuthenticationMethodHTTPDigest ||
-            challenge.protectionSpace.authenticationMethod == NSURLAuthenticationMethodNTLM else {
-      return (.performDefaultHandling, nil)
-    }
-
-    if !domains.contains(challenge.protectionSpace.host) {
-      return (.performDefaultHandling, nil)
+    // Certificate Pinning
```

Comment:
What's the reason to have a separate pinning check here?

Reply:
I have it there in case the API call needs to be pinned.

Reply:
Do we not already pin brave hosts in API calls? The ssl pinning is meant to be for API calls specifically. That's why I'm surprised we have additional code here.

Reply:
We pin API calls for playlist and DAU, but I don't think we had pinning for search APIs as we didn't really make any (the calls go to Google as the fallback). AFAIK, all search calls happened in the webView as the user searches via typing into the page itself. All our WebKit calls are pinned.

Reply:
Maybe not as a part of this PR, but can we enable SSL pinning for other Brave API calls as well? All Web3 calls to our proxies are not pinned, ideally we need to pin all the API calls to Brave domains in https://github.com/brave/brave-core/blob/master/chromium_src/net/tools/transport_security_state_generator/input_file_parsers.cc
cc @StephenBass

Reply:
Ticket is here https://github.com/brave/brave-ios/issues/2440
We will triage it next week